database/oracle/sqlpluscmd.pm

```diff
@@ -0,0 +1,466 @@
+    
+    if (!defined($options{output})) {
+        print "Class sqlpluscmd: Need to specify 'output' argument.\n";
+        exit 3;
+    }
+    if (!defined($options{options})) {
+        $options{output}->add_option_msg(short_msg => "Class sqlpluscmd: Need to specify 'options' argument.");
+        $options{output}->option_exit();
+    }
+    if (!defined($options{noptions})) {
+        $options{options}->add_options(arguments => 
+                    { "sqlplus-cmd:s"            => { name => 'sqlplus_cmd'},
+                      "oracle-sid:s"             => { name => 'oracle_sid' },
+                      "oracle-home:s"           => { name => 'oracle_home' },
+                      "tnsadmin-home:s"         => { name => 'tnsadmin_home' },
+                      "username:s"              => { name => 'username' },
+                      "password:s"              => { name => 'password' },
+                      "local-connexion"       => { name => 'local_connexion', default=>0 }, 
+                      "oracle-debug"       => { name => 'oracle_debug', default=>0 }, 
+                      "sysdba"                => { name => 'sysdba', default=>0 }, 
+                      "sql-errors-exit:s"        => { name => 'sql_errors_exit', default => 'unknown' },
+                      "tempdir:s"        		=> { name => 'tempdir', default => '/tmp' },
+        });
+    }
+    $options{options}->add_help(package => __PACKAGE__, sections => 'sqlpluscmd OPTIONS', once => 1);
+
+    $self->{output} = $options{output};
+    $self->{mode} = $options{mode};
+    $self->{args} = undef;
+    $self->{stdout} = undef;
+    $self->{columns} = undef;
+    $self->{version} = undef;
+    
+    $self->{oracle_sid} = undef;
+    $self->{oracle_home} = undef;
+    $self->{tnsadmin_home} = undef;
+    $self->{local_connexion} = undef;
+    $self->{sysdba} = undef;
+    $self->{username} = undef;
+    $self->{password} = undef;
+    $self->{tempdir} = undef;
+    
+    return $self;
+}
+
+# Method to manage multiples
+sub set_options {
+    my ($self, %options) = @_;
+    # options{options_result}
+
+    $self->{option_results} = $options{option_results};
+}
+
+# Method to manage multiples
+sub set_defaults {
+    my ($self, %options) = @_;
+    # options{default}
+    
+    # Manage default value
+    foreach (keys %{$options{default}}) {
+        if ($_ eq $self->{mode}) {
+            for (my $i = 0; $i < scalar(@{$options{default}->{$_}}); $i++) {
+                foreach my $opt (keys %{$options{default}->{$_}[$i]}) {
+                    if (!defined($self->{option_results}->{$opt}[$i])) {
+                        $self->{option_results}->{$opt}[$i] = $options{default}->{$_}[$i]->{$opt};
+                    }
+                }
+            }
+        }
+    }
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    # return 1 = ok still data_source
+    # return 0 = no data_source left
+
+    $self->{oracle_sid} = $self->{option_results}->{oracle_sid};

```

Comment:
please use directly $self->{option_results}->{xxxx) value no need to redefine directly on self obj